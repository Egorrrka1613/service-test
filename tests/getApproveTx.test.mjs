import { getApproveTx } from "../api/index.mjs";
import { evm } from '@zomet/sdk';
import { jest } from '@jest/globals'
import assertValidate from './helpers/assertValidate.mjs';
import nock from 'nock';

let inputData;
beforeEach(() => {
    nock.cleanAll();
    inputData = {
        net: 'eth',
        tokenAddress: '0xdac17f958d2ee523a2206206994597c13d831ec7',
        ownerAddress: '0xaac594a9b83f892f1a4b5b8804339fc6ddb797ee'
    }
});

afterAll(() => {
    nock.restore();
});

describe('getApproveTx method tests', () => {
    it('Case #1: Get approve tx swap from Ether to USDT in ETH', async () => {
        // given
        const txMock = {
            from: inputData.ownerAddress,
            to: inputData.tokenAddress,
            data: '0x095ea7b30000000000000000000000001111111254eeb25477b68fb85ed929f73a960582000000000000000000000000000000000000000000000000001fffffffffffff',
            value: '0',
            nonce: 0,
            gasPrice: '29568592931',
            gas: '48597',
            chainId: 1
        };

        const inchRouterContractMock = { buildTransaction: jest.fn().mockReturnValue(txMock) };
        jest.spyOn(evm.standards, 'ERC20').mockReturnValue(inchRouterContractMock);

        // when
        const response = await getApproveTx(inputData);

        // then
        expect(inchRouterContractMock.buildTransaction).toHaveBeenCalled();
        expect(response).toMatchSnapshot();
    });

    it(`Case #2: Assert error if use invalid method signature`,
        async () => assertValidate(getApproveTx, inputData));
});
