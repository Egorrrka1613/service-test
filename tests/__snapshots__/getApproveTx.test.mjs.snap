// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getApproveTx method tests Case #1: Get approve tx swap from Ether to USDT in ETH 1`] = `
{
  "chainId": 1,
  "data": "0x095ea7b30000000000000000000000001111111254eeb25477b68fb85ed929f73a960582000000000000000000000000000000000000000000000000001fffffffffffff",
  "from": "0xaac594a9b83f892f1a4b5b8804339fc6ddb797ee",
  "gas": "48597",
  "gasPrice": "29568592931",
  "nonce": 0,
  "to": "0xdac17f958d2ee523a2206206994597c13d831ec7",
  "value": "0",
}
`;

exports[`getApproveTx method tests Case #2: Assert error if use invalid method signature 1`] = `""net" is required"`;

exports[`getApproveTx method tests Case #2: Assert error if use invalid method signature 2`] = `""tokenAddress" is required"`;

exports[`getApproveTx method tests Case #2: Assert error if use invalid method signature 3`] = `""ownerAddress" is required"`;
